src/tnetdbg.hpp

```diff
@@ -56,9 +58,12 @@ extern const bool g_is_windows_console;
 
 void write_to_console(const std::string& obj);
 
+extern std::recursive_mutex _g_dbg_mutex;
+
 #define DBGLVL(N) if (!(N>=g_dbg_level)) break
+#define DBGLOCK std::lock_guard<std::recursive_mutex> _dbg_lock( _g_dbg_mutex );
 
-#define _main_dbg(X) std::ostringstream _dbg_oss; _dbg_oss << X; write_to_console(_dbg_oss.str())
+#define _main_dbg(X) { DBGLOCK; std::ostringstream _dbg_oss; _dbg_oss << X; write_to_console(_dbg_oss.str()); }
```

Comment:
We need reserved names.
Can't do just info() as it would debug with our own code  and other libs.

We could do $info() and such I guess....